components/user/UserProfile.js

```diff
@@ -23,13 +25,38 @@ function UserProfile({ BASE_URL, data }) {
 
   //Declared Ref object for QR
   const qrRef = useRef(null);
+  const elementRef = useRef(null);
 
   //qrRef.current is pointing to the DOM node and firstChild to its canvas
   const downloadQR = () =>
     qrRef.current.firstChild.toBlob((blob) =>
       saveAs(blob, `biodrop-${data.username}.png`),
     );
 
+  const downloadWallpaper = async () => {
+    try {
+      // It removes the "hidden" class to make the element visible for rendering it.
+      elementRef.current.classList.remove("hidden");
+
+      const dataUrl = await toPng(elementRef.current, {
+        cacheBust: false,
+        backgroundColor: "#122640",
+        width: 1080,
+        height: 1920,
+      });
+
+      const link = document.createElement("a");
+      link.download = "Biodrop-QRCode-Wallpaper.png";
+      link.href = dataUrl;
+      link.click();
+
+      // It adds the "hidden" class back to hide the element again.
+      elementRef.current.classList.add("hidden");
+    } catch (e) {
+      console.log(e);
```

Comment:
To be replaced with `Logger`

Reply:
Did you try from an existing example `pages/account/manage/profile.js`

Reply:
Whenever I use Logger in the file. I get logged out. I have attached the screenshot below:

WIth `console.log` in the file
![consolelogin](https://github.com/EddieHubCommunity/BioDrop/assets/72060440/0a968845-a53f-43eb-8a49-681acaaa628d)

Whenever I replace `console` with `logger `in the file, it changes _Account_ into _Login_. Then when I presss the FaShare icon it throws _ADMIN_USERS NOT FOUND_(may be the _"Account-to-Login Change"_ has something to do with the error")
![loggererror](https://github.com/EddieHubCommunity/BioDrop/assets/72060440/d1331c9d-c1a6-4adf-8fc4-8f755d0a4d92)